backend/services/project_service.py

```diff
@@ -615,3 +617,37 @@ def send_email_on_project_progress(project_id):
                     project_completion,
                 ),
             ).start()
+
+    @staticmethod
+    def get_active_projects(interval):
+        action_date = datetime.now(timezone.utc) - timedelta(hours=interval)
+        result = (
+            TaskHistory.query.with_entities(TaskHistory.project_id)
+            .distinct()
+            .filter(TaskHistory.action_date >= action_date)
```

Comment:
Using this , test this query with dump of production database , you might need action_date btree timestamp index